### PR TITLE
invoke Capture() when slice elements implement Capture interface

### PR DESCRIPTION
**Pull request by daeMOn63**
Maybe a fix option for #122 

Works on the issue POC, but still not 100% sure on potential side effects and issues.

### CODE REVIEW

**Comment by alecthomas** [comment]
Great idea!

You'll need to support the `Parseable` interface too though. I think there should be some common code you can call here, but if not you'll have to factor it out of the existing code.

Thanks!

**Comment by daeMOn63** [comment]
Just tried to reproduce the issue using a  `Parseable` type (see https://play.golang.org/p/donM1qN1UU5), and it properly calls `Parse` for each struct members and slice elements.

**Comment by alecthomas** [comment]
Huh ok, that's a bit surprising. If you could add a test that would be excellent, then I'll merge. Thanks again.

**Comment by daeMOn63** [comment]
Awesome thanks you! 

I've added a couple of test for both interfaces (Parseable and Capture)

I've also discovered another unrelated issue on the way, where a type cannot implement both `Parseable` and `Capture` at the same time, otherwise we get a panic on https://github.com/alecthomas/participle/blob/master/nodes.go#L591, as the field value cannot be converted to a string. Maybe could be fixed adding a `String() string` to the `Capture` interface, but it would bc break things... Lmk if you'd like another open issue to keep track of this one :)

Also had golint getting in the way, reporting test code as duplicate, for reasons I don't understand yet. Have ignored the check for now, but let me know if anything better can be done.

File: nodes.go [hunk 1]
```diff
@@ -576,11 +576,20 @@ func setField(tokens []lexer.Token, strct reflect.Value, field structLexerField,
 	}
 
 	if f.Kind() == reflect.Slice {
-		fieldValue, err = conform(f.Type().Elem(), fieldValue)
-		if err != nil {
-			return err
+		if d, ok := reflect.New(f.Type().Elem()).Interface().(Capture); ok {
```
Comment (alecthomas) [inline nodes.go:579]: This will always create a new element, even if the resulting value doesn't implement `Capture`. Please refactor to avoid this, perhaps using `reflect.Type.Implements`.

File: nodes.go [hunk 2]
```diff
@@ -576,11 +576,20 @@ func setField(tokens []lexer.Token, strct reflect.Value, field structLexerField,
...
+			for _, v := range fieldValue {
+				if err := d.Capture([]string{v.Interface().(string)}); err != nil {
```
Comment (alecthomas) [inline nodes.go:581]: I don't think this code is correct. You should be constructing a new `d` for each element appended to `f`. I think this is only working in the tests at the moment because each `d` is a value and gets copied. If `d` were a pointer to a struct, for example, it would get repeatedly overwritten.

**Review by alecthomas — CHANGES_REQUESTED** [review]

File: nodes.go (hunk 1 above)
Comment (daeMOn63) [reply nodes.go]: right, thanks! fixed now

File: nodes.go (hunk 2 above)
Comment (daeMOn63) [reply nodes.go:586]: Indeed, it should be fixed now.

**Comment by alecthomas** [comment]
Brilliant, thanks :)
